src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -253,242 +253,105 @@ static_assert(MAV_SENSOR_ROTATION_ROLL_90_PITCH_68_YAW_293 == static_cast<MAV_SE
 static_assert(MAV_SENSOR_ROTATION_PITCH_315 == static_cast<MAV_SENSOR_ORIENTATION>(ROTATION_PITCH_315), "Pitch: 315");
 static_assert(MAV_SENSOR_ROTATION_ROLL_90_PITCH_315 == static_cast<MAV_SENSOR_ORIENTATION>(ROTATION_ROLL_90_PITCH_315),
 	      "Roll: 90, Pitch: 315");
-
 static_assert(41 == ROTATION_MAX, "Keep MAV_SENSOR_ROTATION and PX4 Rotation in sync");
 
-void get_mavlink_navigation_mode(const struct vehicle_status_s *const status, uint8_t *mavlink_base_mode,
-				 union px4_custom_mode *custom_mode)
-{
-	custom_mode->data = 0;
-	*mavlink_base_mode = 0;
-
-	/* HIL */
-	if (status->hil_state == vehicle_status_s::HIL_STATE_ON) {
-		*mavlink_base_mode |= MAV_MODE_FLAG_HIL_ENABLED;
-	}
-
-	/* arming state */
-	if (status->arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
-		*mavlink_base_mode |= MAV_MODE_FLAG_SAFETY_ARMED;
-	}
-
-	/* main state */
-	*mavlink_base_mode |= MAV_MODE_FLAG_CUSTOM_MODE_ENABLED;
 
-	const uint8_t auto_mode_flags	= MAV_MODE_FLAG_AUTO_ENABLED
-					  | MAV_MODE_FLAG_STABILIZE_ENABLED
-					  | MAV_MODE_FLAG_GUIDED_ENABLED;
+union px4_custom_mode get_px4_custom_mode(uint8_t nav_state)
+{
+	union px4_custom_mode custom_mode;
+	custom_mode.data = 0;
 
-	switch (status->nav_state) {
+	switch (nav_state) {
 	case vehicle_status_s::NAVIGATION_STATE_MANUAL:
-		*mavlink_base_mode	|= MAV_MODE_FLAG_MANUAL_INPUT_ENABLED
-					   | (status->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING ? MAV_MODE_FLAG_STABILIZE_ENABLED : 0);
```

Comment:
Did you remove this logic on purpose?

Reply:
This part is now in the HEARTBEAT, but set from `vehicle_control_mode` which already handles these VTOL specifics in commander.